hudi-spark-datasource/hudi-spark3.2plus-common/pom.xml

```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>hudi-spark-datasource</artifactId>
+        <groupId>org.apache.hudi</groupId>
+        <version>0.13.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>hudi-spark3.2plus-common</artifactId>
```

Comment:
we can't be sure if 3.4 and beyond can still share these common classes, can we? then are we gonna rename this module if this does not apply to 3.4?

Reply:
Not so clear cut unfortunately: we have to go case-by-case in some cases we will need to move classes out in some cases we might need to duplicate. 

Usually though, it's not "regressing" -- in a sense that it's more likely that 3.4 will be mostly compatible w/ 3.3, than w/ 3.3 in which case we just introduce new `spark3.3plus` module

Reply:
yes. the other option also does not work smoothly either. for eg, we could name this spark3.2_3.3. what incase spark3.4 is compatible. either ways, we have to rename some of these when we have more patched version or minor versions. probably we can go w/ what we have in this patch.

Reply:
ok sounds good to me.